pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -96,32 +134,39 @@ func (q *QueryReconciler) collectInfoFromVdb(ctx context.Context) (ctrl.Result,
 				return e
 			}
 		}
-		// extract out the communal and config information to pass down to the vclusterops API.
-		opts = append(opts,
-			restorepointsquery.WithCommunalPath(vdb.GetCommunalPath()),
-			restorepointsquery.WithConfigurationParams(q.ConfigurationParams.GetMap()),
-		)
-
 		return nil
 	})
 
 	return res, err
 }
 
-// runListRestorePoints will update the status condition before and after calling
-// list restore points api
-// Temporarily, runListRestorePoints will not call the ListRestorePoints API
-// since the dispatcher is not set up yet
-func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
+// runShowRestorePoints will update the status condition and state before and after calling
+// showrestorepoints vclusterops api
+func (q *QueryReconciler) runShowRestorePoints(ctx context.Context, _ *ctrl.Request, dispatcher vadmin.Dispatcher,
+	opts []restorepointsquery.Option) (err error) {
 	// set Querying status condition and state prior to calling vclusterops API
-	err := vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
 		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, "Started"), stateQuerying)
 	if err != nil {
 		return err
 	}
 
-	// API should be called to proceed here
-	// If we receive a failure result from the API, a state message and condition need to be updated
+	// call showRestorePoints vcluster API
+	q.VRec.Eventf(q.Vrpq, corev1.EventTypeNormal, events.ShowRestorePointsStarted,
+		"Starting show restore points")
+	start := time.Now()
+	if res, errRun := dispatcher.ShowRestorePoints(ctx, opts...); verrors.IsReconcileAborted(res, errRun) {
+		if errRun != nil {
+			err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+				v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "Failed"), stateFailedQuery)
+			if err != nil {
+				return err
```

Comment:
Lets join the two errors rather than losing one.
```suggestion
				errRun = errors.Join(errRun, err)
```